[feat] 공간-문화 뷰 UI 구현

seungjunGong opened the pull request:
## Related issue 🛠
- closed #16

## Work Description 📝
- 공간 문화뷰 구현
- what's on contents 컴포넌트
- space, culture 카드 컴포넌트 구현

## Screenshot 📸
https://github.com/user-attachments/assets/a9910550-9bf1-4d70-bf99-152f93e4afcf


## Uncompleted Tasks 😅
N/A

## To Reviewers 📢
what's on model 은 임시로 presentation 모델에 두었습니다.
로고 svg -> png 로 변경 했더니 해결 되었어요😅

로고 클릭시 무조건 detail 로 가게 두었는데 culture 부분은 막아둘까요??

seungjunGong committed:
[feat] #16 Whatson 리스트 아이템 구현

seungjunGong committed:
[feat] #16 Whatson 리스트 아이템 예매 추가

seungjunGong committed:
[chore] #16 WhatsOnItem 레이아웃 수정

seungjunGong committed:
[feat] #16 Whatson 리스트 아이템 예매 추가

seungjunGong committed:
[chore] #16 WhatsOnListItem 이름 변경

seungjunGong committed:
[chore] #16 SpaceAndCultureHeader line 색상 수정

seungjunGong committed:
[feat] #16 Whats on 구현 완료

seungjunGong committed:
[feat] #16: card info 추가

seungjunGong committed:
[chore] #16: SpaceAndCultureHeader 설명 color 수정

seungjunGong committed:
[chore] #16: 로고 png 변경

seungjunGong committed:
[feat] #16: space, culture grid 구현

seungjunGong committed:
[chore] #16: ktlint 적용

MinseoSONG commented on app/src/main/java/org/sopt/hyundaicarddive/presentation/type/SpaceAndCultureGridItem.kt:
깔쌈하고 예뻐요잉

근데 생각해보니까 너 말대로 SpaceAndCultureGridItem을 기반으로 Space랑 Culture를 각각 만들어주는게 더 이뻤을 거 같기도 하고 .. ? ㅎㅎ ..
이런 오비라 미안하다 ! 지만 지금도 이뻐서 굳이 리팩할 필요는 없어용 하고 싶다면 해 !!! 내 잘못임 ㅠㅠ !!!
![image](https://github.com/user-attachments/assets/ee94eaa4-8440-4be3-bccf-a6df07838c1f)
변명하자면 .. 물어볼 때 .. 나 솝커톤 기초세팅 중이엇어서 넘 정신없엇어 .. 리팩할거면 하고나서 또 불러줘 ...... 무조건 정성 코리할게 . . 🥲

seungjunGong replied:
오류 나서 나도 정신 없었어ㅋㅋㅋ 바쁜데도 봐주고 고맙데이
이대로도 좋아서 수정 없이 갈게요

MinseoSONG commented on app/src/main/java/org/sopt/hyundaicarddive/presentation/type/SpaceAndCultureGridItem.kt:
```diff
@@ -0,0 +1,75 @@
+        logoResId = R.drawable.img_space_logo_designlib
+    ),
+    SPACE_MUSICLIB(
+        R.string.spaceandculture_space_card_musiclib,
+        R.drawable.img_space_logo_musiclib
```
그리고 ! 그때도 말했지만 명시적으로 어떤 속성에 어떤 값을 넣었는지 표기해주는게 좋음~

위처럼 description = ~, logoResId = ~이렇게 해두면 다른 개발자가 이 코드를 봐도 아 여기에 이 값이 들어갔구나 굳이 spaceAndCultureGridItem의 속성이 뭔지 위에 올라가서 확인할 필요가 없이 바로 알 수 있으니깐뇨

MinseoSONG commented on app/src/main/java/org/sopt/hyundaicarddive/presentation/ui/spaceandculture/SpaceAndCultureScreen.kt:
```diff
@@ -1,33 +1,106 @@
-        text = "SpaceAndCultureScreen",
-        modifier = Modifier.clickable { navigateToDetail() }
-    )
+    Column(
+        modifier = modifier.fillMaxSize()
+    ) {
+        TopBar(
+            topBarType = TopBarType.SPACEANDCULTURE
+        )
+        LazyColumn(
+            contentPadding = padding
+        ) {
```
상위 Column에 padding을 안주시고 하위 LazyColum에 padding을 주신 이유가 있을까요?? 

지금 코드로면 topBar에는 padding값ㅇ ㅣ 적용되지 않는데, 현재 저희 코드 상 모든 화면에 MainScreen, MainNavHost를 통해 동일한 padding 값을 주고 있기 때문에 Screen에 바로 적용해주는 것이 좋을 것 같습니다 ~~

```
@Composable
fun HomeRoute(
    padding: PaddingValues,
    viewModel: HomeViewModel = hiltViewModel()
) {
    LaunchedEffect(Unit) {
        viewModel.getHomeList()
    }
    val selectedOption by viewModel.selectedOption.collectAsStateWithLifecycle()
    val selectedCategory by viewModel.selectedCategory.collectAsStateWithLifecycle()
    val changeListAlign by viewModel.changeListAlign.collectAsStateWithLifecycle()
    val homeList by viewModel.homeList.collectAsStateWithLifecycle()

    HomeScreen(
        selectedOption = selectedOption,
        onOptionSelected = { viewModel.onOptionSelected(it) },
        selectedCategory = selectedCategory,
        onCategorySelected = { viewModel.onCategorySelected(it) },
        changeListAlign = changeListAlign,
        onChangeListAlign = { viewModel.onChangeListAlign() },
        homeList = homeList,
        modifier = Modifier.padding(padding)
    )
}

@Composable
private fun HomeScreen(
    selectedOption: Int,
    onOptionSelected: (Int) -> Unit,
    selectedCategory: Int,
    onCategorySelected: (Int) -> Unit,
    changeListAlign: Boolean,
    onChangeListAlign: () -> Unit,
    homeList: List<HomeListModel>,
    modifier: Modifier = Modifier
) {
    Box(
        modifier = modifier.fillMaxSize()
    ) {
```
위는 제 홈화면 코드인데 저 같은 경우는, 받아온 padding 값을 HomeScreen의 modifier에 바로 적용을 해줘서 가장 상위에 있는 Box에 modifier를 적용해주었답니ㄷ ㅏ!
참고해서 수정해보시면 좋을 것 같아요 ~

seungjunGong replied:
헉 그러네요.. topbar에는 적용이 안되겠네요ㅜ
수정 하겠습니다 !!

MinseoSONG commented on app/src/main/java/org/sopt/hyundaicarddive/presentation/ui/spaceandculture/SpaceAndCultureViewModel.kt:
현재 spaceList와 cultureList는 서버에서 받아오는 값이 아니라 고정된 데이터인데도 StateFlow로 상태처럼 관리하고 get~함수를 통해 초기화하고 있네요 ~ 
하지만 이 방식은 불필요하게 상ㅌㅐ를 관리하고 로직을 분산시켜 코드 복잡도를 더 높이고 있는 것 같습니다 ..
어차피 서버에서 안받아오고 로컬에 박을 값이라면 
```
val spaceList = listOf(
    ...
)

val cultureList = listOf(
    ...
)

```
이런 식으로 고정 리스트는 viewmodel 외부에서 직접 참조 가능하게 정의하거나, 그냥 val로 노출하는게 유지보수와 가독성 측면에서 더 좋을 것 같습니다 !

seungjunGong replied:
네 저도 그부분이 맘에 걸렸었는데 .. 직접 참조하게 하는 방식이 더 나아보이네요 !! "이 방식은 불필요하게 상ㅌㅐ를 관리하고 로직을 분산시켜 코드 복잡도를 더 높이고 있는 것" 상세한 설명 감사합니다.

MinseoSONG reviewed (approved):
수고햇다 !!!!!
culture 부분 막아두는거 찬성이에욥
코리 남겨둔거 보고 모르는거 있으면 질문해주세용 글고 코리 반영 다 하면 바로 머지해도 좋을 것 같아 미리 어푸 찍어둠 !
![image](https://github.com/user-attachments/assets/89c29d7b-be4b-47a2-b877-0b1c0aea27b5)
수고햇다 복복

seungjunGong committed:
Merge branch 'develop' of https://github.com/SOPT-all/36-COLLABORATION-ANDROID-HYUNDAICARDDIVE into feat-space-culture-ui

seungjunGong committed:
[chore] #16 SpaceAndCulture padding 및 구조 정리

- SpaceAndCultureScreen padding 수정
- SpaceAndCultureViewModel logo list 참조 방식 수정
- SpaceAndCultureGridItem 속성 명시

seungjunGong committed:
[chore] #16: culture detail 네비 제거, ktlint 적용

mjeong21 commented on app/src/main/java/org/sopt/hyundaicarddive/presentation/ui/spaceandculture/SpaceAndCultureScreen.kt:
```diff
@@ -1,42 +1,138 @@
-        padding = padding,
-        navigateToDetail = navigateToDetail
+        navigateToDetail = navigateToDetail,
+        whatsOnList = whatsOnList,
+        spaceList = viewModel.spaceListItems,
+        cultureList = viewModel.cultureListItems,
```
데이터만 넘기기 조와용!

mjeong21 commented on app/src/main/java/org/sopt/hyundaicarddive/presentation/ui/spaceandculture/component/SpaceAndCultureCardGrid.kt:
```diff
@@ -0,0 +1,115 @@
+
+@Composable
+fun SpaceAndCultureCardGrid(
+    modifier: Modifier = Modifier,
+    cards: List<SpaceAndCultureGridItem>,
```
여기 Modifier 앞에는 필수 인자, 뒤에는 선택 인자를 작성하는 것으로 배웠는데 순서를 바꾸는 건 어떨까용?

mjeong21 reviewed (approved):
drawable 파일 추가하느라 고생 많았습니당 ㅜㅜ!!!

seungjunGong committed:
Merge branch 'develop' of https://github.com/SOPT-all/36-COLLABORATION-ANDROID-HYUNDAICARDDIVE into feat-space-culture-ui

seungjunGong committed:
[chore] #16: SpaceAndCultureCardGrid 파라미터 순서 변경
